src/v/cloud_topics/frontend/frontend.cc

```diff
@@ -455,6 +494,32 @@ ss::future<std::expected<kafka::offset, std::error_code>> frontend::replicate(
         co_return std::unexpected(res.error());
     }
 
+    auto fence_fut = co_await ss::coroutine::as_future(
+      _ctp_stm_api->fence_epoch(res.value().front().id.epoch));
+    if (fence_fut.failed()) {
+        // TODO: handle shutdown failures gracefully
+        auto e = fence_fut.get_exception();
+        vlog(
+          cd_log.warn,
+          "Failed to fence epoch {} for ntp {}, error: {}",
+          res.value().front().id.epoch,
+          ntp(),
+          fence_fut.get_exception());
+        std::rethrow_exception(e);
+    }
+    auto fence = std::move(fence_fut.get());
```

Comment:
I'm assuming there will be a followup to use the `term` in the partition replicate call?